pkg/virt-handler/rest/console.go

```diff
@@ -349,6 +349,8 @@ func (t *ConsoleHandler) stream(vmi *v1.VirtualMachineInstance, request *restful
 	select {
 	case <-stopCh:
 		break
+	case <-request.Request.Context().Done():
```

Comment:
Hi @orelmisan 
disregarding the rest of this PR,
this line might be interesting, it allows to directly close a connection, instead waiting for timeout

changes on tests/console/login.go might be worth as well, they can improve the retry mechanism (but in case there is indeed a bug about closing and reopening connections, it will get worse unless the above mention fix, is the missing link

it still wont solve why VM doesnt even start sometimes, but this is other issue of course